core/block_context.go

```diff
@@ -0,0 +1,87 @@
+package core
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/contract_comm/blockchain_parameters"
+	"github.com/celo-org/celo-blockchain/contract_comm/currency"
+	gpm "github.com/celo-org/celo-blockchain/contract_comm/gasprice_minimum"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+)
+
+// BlockContext represents contextual information about the blockchain state
+// for a given block
+type BlockContext interface {
```

Comment:
Having the interface and implementation is a bit of an anti-pattern in Go. Combined with the fact that there is only one implementation, I would suggest only defining a `type BlockContext struct` here and defining an interface with the consuming code if needed. (This is the suggested approach [in Effective Go](https://golang.org/doc/effective_go#interfaces))

Reply:
done